cluster-autoscaler/expander/factory/expander_factory.go

```diff
@@ -48,12 +55,30 @@ func ExpanderStrategyFromString(expanderFlag string, cloudProvider cloudprovider
 			price.NewSimplePreferredNodeProvider(autoscalingKubeClients.AllNodeLister()),
 			price.SimpleNodeUnfitness), nil
 	case expander.PriorityBasedExpanderName:
-		maps := kubeClient.CoreV1().ConfigMaps(configNamespace)
-		initialPriorities, priorityChangesChan, err := priority.InitPriorityConfigMap(maps, configNamespace)
-		if err != nil {
-			return nil, errors.ToAutoscalerError(errors.InternalError, err)
-		}
-		return priority.NewStrategy(initialPriorities, priorityChangesChan, autoscalingKubeClients.LogRecorder)
+		// TODO: how to get proper termination info? It seems other listers do the same here
+		// they never receive the termination msg on the ch
+		stopChannel := make(chan struct{})
+		lister := getPriorityExpandersConfigMapLister(kubeClient, stopChannel, configNamespace)
+		return priority.NewStrategy(lister.ConfigMaps(configNamespace), autoscalingKubeClients.Recorder)
 	}
 	return nil, errors.NewAutoscalerError(errors.InternalError, "Expander %s not supported", expanderFlag)
 }
+
+func getPriorityExpandersConfigMapLister(kubeClient kube_client.Interface, stopChannel <-chan struct{},
+	namespace string) v1lister.ConfigMapLister {
+	// FIXME: how to be sure the reflector completed at least one run and switch to the line below?
```

Comment:
@MaciekPytel I have a problem here: how to make sure the reflector completed at least 1 sync loop before I access the lister? I want to validate the configmap before starting.
If this can't be done easily, I probably have to give up on startup time validation and assume reflector will be ready when first call for `Best` comes. But even that smells like race condition.

Reply:
CA loop generally takes a few seconds before it gets to expander and other listers must have already synced for a scale-up to have any chance of happening. Since you have a fallback strategy and the race is quite unlikely I'm ok to merge this without logic to explicitly make sure lister has initialized. 
If you think you really need to make sure: I think the LastSyncResourceVersion() you do below is the best way unfortunately (at least the best way I know of). I don't like creating lister in here though. This file is really not the place for all those client-go internals. If you really want to sync, I'd rather you created your own configmap lister interface in utils/kubernetes/listers.go (like we do for pod listers) and expose a HasSynced() method.